src/item.cpp

```diff
@@ -4309,60 +4309,61 @@ const item * item::gunmod_find( const itype_id& mod ) const
     return const_cast<item *>( this )->gunmod_find( mod );
 }
 
-bool item::gunmod_compatible( const item& mod, bool alert, bool effects ) const
+bool item::gunmod_compatible( const item& mod, std::string *msg ) const
```

Comment:
All arguments should have a similar name. `msg` or `err` are both acceptable but not a mix of both.